Sources/Purchasing/ProductsManager.swift

```diff
@@ -126,6 +135,25 @@ private extension ProductsManager {
         return self.productsFetcherSK1.products(withIdentifiers: identifiers, completion: completion)
     }
 
+    func trackProductsRequestIfNeeded(_ startTime: Date,
+                                     storeKitVersion: StoreKitVersion,
+                                     error: PurchasesError?) {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
+           let diagnosticsTracker = self.diagnosticsTracker {
+            let responseTime = self.dateProvider.now().timeIntervalSince(startTime)
+            let errorMessage = (error?.userInfo[NSUnderlyingErrorKey] as? Error)?.localizedDescription
+                ?? error?.localizedDescription
+            let errorCode = error?.errorCode
+            Task(priority: .background) {
+                await diagnosticsTracker.trackProductsRequest(wasSuccessful: error == nil,
```

Comment:
wonder whether the diagnostics tracker should be the one to know that its stuff is all background, instead of every consumer needing to know that

Reply:
That's a valid point... And yeah, you might be right. I will change this one and the other ones we're sending, so it's encapsulated. Thanks!

Reply:
I did this in https://github.com/RevenueCat/purchases-ios/pull/4270